### PR TITLE
Season stat refact

### PR DESCRIPTION
**Pull request by sagemlee**
-Added refactors to season statistics
-updated stat tracker with new method names

### CODE REVIEW

File: lib/season_statistics.rb [hunk 1]
```diff
@@ -1,194 +1,141 @@
...
   end
 
-  def teams_hash(season)
-    teams = Hash.new(0)
-    season_teams = current_season_game_teams(season).map do |game|
-      game.team_id
+
+  #returns an array of game_team objects in given season
+    def current_season_game_teams(season)
+      season_games = current_season_game_ids(season)
```
Comment (sagemlee) [inline lib/season_statistics.rb:24]: Still needs to be refactored with memo

File: lib/season_statistics.rb [hunk 2]
```diff
@@ -1,194 +1,141 @@
...
+    season_game_ids.compact
   end
 
-  def teams_hash(season)
-    teams = Hash.new(0)
-    season_teams = current_season_game_teams(season).map do |game|
-      game.team_id
+
+  #returns an array of game_team objects in given season
+    def current_season_game_teams(season)
```
Comment (takeller) [inline lib/season_statistics.rb:23]: I think this is where our performance issue is coming from. We call this method about 7 times per season, and each time is iterates over the game_teams_collection. We will have to come back to this.

File: lib/season_statistics.rb [hunk 3]
```diff
@@ -1,194 +1,141 @@
 require 'pry'
 
 class SeasonStatistics
-attr_reader :stat_tracker, :game_collection, :game_teams_collection, :teams_collection
+attr_reader :game_collection
```
Comment (sagemlee) [inline lib/season_statistics.rb:4]: Initialize can be moved to parent class

**Review by takeller — APPROVED** [review]
This all looks great. I like all of this refactoring. I left an inline comment where I think our performance issue is coming from.

**Review by jhgould — APPROVED** [review]

**Pull request merged**